### ISSUE TITLE
When alternative function has same name as disallowed function, using the alternative function is reported as an error in the file where the latter is defined

### ISSUE THREAD

**Issue by ZebulanStanphill** [spaze/phpstan-disallowed-calls#118]
This one is a bit tough to explain, so here's a code example:

```php
<?php declare(strict_types=1);
namespace MyNamespace;

use function __ as wp__;

function __(string $str): string {
	return wp__($str, 'foo');
}

function someOtherFn(): string {
	return __('whatever'); // The __ used here is MyNamespace\__
}
```

```neon
parameters:
	disallowedFunctionCalls:
		-
			function: '__'
			message: 'use MyNamespace\__ instead'
```

When running PHPStan, 2 usages of `__` are reported, even though there's only 1 (the other is actually `MyNamespace\__`. This issue only occurs in the file where the `MyNamespace\__` is defined. In the above example, I even renamed `__` for the file, but that didn't help.

**Comment by spaze** [comment]
Oh! Nice catch, thanks for the report! And thanks for the code, it certainly helped me to understand the problem and develop a fix which will be landing soon.

**Comment by spaze** [comment]
Fixed and released as [2.6.0](https://github.com/spaze/phpstan-disallowed-calls/releases/tag/v2.6.0), let me know if it works in your case.

**Comment by ZebulanStanphill** [comment]
The fix worked, but it also introduced a regression: #124.

**Comment by spaze** [comment]
Oh no... Thanks for spotting that and sorry. I've pulled the 2.6.0 release and will fix and release 2.6.1 after a bit of 😴

And I thought I have enough tests.

**Comment by spaze** [comment]
Just to close the loop here as well, the fix (#126) was released in [2.7.0](https://github.com/spaze/phpstan-disallowed-calls/releases/tag/v2.7.0), hopefully no other problem introduced.

**Comment by ZebulanStanphill** [comment]
Thanks for all your help. Unfortunately, it looks like #126 ended up bringing back the original problem reported here.
